salt/modules/zypperpkg.py

```diff
@@ -1171,22 +1171,39 @@ def mod_repo(repo, **kwargs):
     return repo
 
 
-def refresh_db():
+def refresh_db(root=None, force=None):
     '''
-    Force a repository refresh by calling ``zypper refresh --force``, return a dict::
+    Trigger a repository refresh by calling ``zypper refresh``. Refresh will run
+    with ``--force`` if the "force=True" flag is passed on the CLI or
+    ``refreshdb_force`` is set to ``true`` in the pillar. The CLI option
+    overrides the pillar setting.
+
+    It will return a dict::
 
         {'<database name>': Bool}
 
     CLI Example:
 
     .. code-block:: bash
 
-        salt '*' pkg.refresh_db
+        salt '*' pkg.refresh_db [force=true|false]
+
+    Pillar Example:
+
+    .. code-block:: yaml
+
+       zypper:
+         refreshdb_force: false
     '''
     # Remove rtag file to keep multiple refreshes from happening in pkg states
     salt.utils.pkg.clear_rtag(__opts__)
     ret = {}
-    out = __zypper__.refreshable.call('refresh', '--force')
+    refresh_opts = ['refresh']
+    if force is None:
+        force = __pillar__.get('zypper', {}).get('refreshdb_force', True)
+    if force:
+        refresh_opts.append('--force')
+    out = __zypper__(root=root).refreshable.call(*refresh_opts)
```

Comment:
Oh, wait a sec. I think the root kwarg is not yet introducted here. Could you check that please?

Reply:
You are correct, that will be introduced with https://github.com/saltstack/salt/pull/54954